lib/skate_web/controllers/test_group_controller.ex

```diff
@@ -0,0 +1,118 @@
+defmodule SkateWeb.TestGroupController do
+  @moduledoc """
+  Provides an interface to view, create, and delete test groups, as well as add
+  and remove users from test groups.
+  """
+
+  use SkateWeb, :controller
+
+  alias Skate.Settings.TestGroup
+  alias Skate.Settings.User
+
+  @spec index(Plug.Conn.t(), map()) :: Plug.Conn.t()
+  def index(conn, _params) do
+    test_groups = TestGroup.get_all() |> Enum.sort_by(& &1.name)
+
+    conn
+    |> assign(:test_groups, test_groups)
+    |> put_layout({SkateWeb.LayoutView, "test_groups.html"})
+    |> render("index.html")
+  end
+
+  @spec post(Plug.Conn.t(), map()) :: Plug.Conn.t()
+  def post(conn, params) do
+    TestGroup.create(params["name"])
+
+    redirect(conn, to: SkateWeb.Router.Helpers.test_group_path(conn, :index))
+  end
+
+  @spec show(Plug.Conn.t(), map()) :: Plug.Conn.t()
+  def show(conn, params) do
+    test_group = TestGroup.get(params["id"])
+
+    if test_group do
+      test_group_users =
+        test_group.users
+        |> Enum.map(&{&1.id, String.downcase(&1.email)})
```

Comment:
nitpick(non-blocking): Emails are down-cased on insert, so this step can be skipped here